File: library/src/main/java/me/proxer/library/entity/messenger/ConferenceParticipant.java
package me.proxer.library.entity.messenger;

import com.squareup.moshi.Json;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Value;
import me.proxer.library.entity.ProxerIdItem;
import me.proxer.library.entity.ProxerImageItem;

import javax.annotation.Nullable;

/**
 * Entity that represents a participant in a {@link Conference}.
 *
 * @author Desnoo
 */
@Value
@EqualsAndHashCode(onParam = @__({@Nullable}))
public class ConferenceParticipant implements ProxerIdItem, ProxerImageItem {

    /**
     * {@inheritDoc}
     */
    @Getter(onMethod = @__({@Override}))
    @Json(name = "uid")
    private String id;

    /**
     * {@inheritDoc}
     */
    @Getter(onMethod = @__({@Override}))
    @Json(name = "avatar")
    private String image;

    /**
     * Returns the username.
     */
    @Json(name = "username")
    private String username;

    /**
     * Returns the current status.
     */
    @Json(name = "status")
    private String status;
}

File: library/src/main/kotlin/me/proxer/library/entity/messenger/ConferenceParticipant.kt
package me.proxer.library.entity.messenger

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import me.proxer.library.entity.ProxerIdItem
import me.proxer.library.entity.ProxerImageItem

/**
 * Entity that represents a participant in a [Conference].
 *
 * @property username The username.
 * @property status The current status.
 *
 * @author Desnoo
 */
@JsonClass(generateAdapter = true)
data class ConferenceParticipant(
    @Json(name = "uid") override val id: String,
    @Json(name = "avatar") override val image: String,
    @Json(name = "username") val username: String,
    @Json(name = "status") val status: String
) : ProxerIdItem, ProxerImageItem
